docs: kraken account type, future trades, margin trades (#6894)
rotki/rotki: docs/usage_guide.rst

Crypto to crypto trades
^^^^^^^^^^^^^^^^^^^^^^^^^^^

A setting to determine whether crypto to crypto trades are taxable and should be taken into account. By default it's ``True``.

Illustration: You have ``1`` ETH that you bought with price ``50 EUR``. Then you spend this ``1`` ETH for gas fees, and the price at the moment is ``100 EUR``.

- ``True``: When you spending crypto (as in fees), a sell is also generated, and this sell can generate additional :green:`profit` or :red:`loss`. Each crypto to crypto trade creates a "virtual" trade that sells or buys the crypto asset for fiat and then sells or buys the fiat for the other crypto asset.

..

    Profit of this sell is ``100 - 50`` = ``50 EUR``.

    Because you spend this ETH, you also sell this ETH for ``100 EUR``, which means you lost ``100 EUR``.
    So total PnL of this action is ``50 - 100`` EUR = ``-50 EUR``

- ``False``: The above does not happen.

..

>     So total PnL of this action will be calculated as ``100 - 50`` = ``50 EUR``.

Author: I think I make mistake in this part. It should be just `-100EUR` instead. Will fix it

A setting to determine whether trade fees should be included into the cost basis of the asset sold/bought or not. By default it's ``True``.

Illustration: You bought ``1`` ETH for ``10 EUR`` and paid ``1 EUR`` fee

- ``True``:  The fee event just reduces the amount of the fee asset paid and nothing else. The actual fee is used to determine the cost basis of the asset sold/bought instead.

> ..

Author: This is for "Include fees in cost basis" setting. I just move the example action, so we don't repeat same thing.

--------------------
A list of supported locations in rotki are ``"external"``, ``"kraken"``, ``"poloniex"``, ``"bittrex"``, ``"binance"``, ``"bitmex"``, ``"coinbase"``, ``"banks"``, ``"blockchain"``, ``"coinbasepro"``, ``"gemini"``, ``"equities"``, ``"realestate"``, ``"commodities"``, ``"cryptocom"``, ``"uniswap"``, ``"bitstamp"``, ``"binanceus"``, ``"bitfinex"``, ``"bitcoinde"``, ``"iconomi"``, ``"kucoin"``, ``"balancer"``, ``"loopring"``, ``"ftx"``, ``"nexo"``, ``"blockfi"``, ``"independentreserve"``, ``"gitcoin"``, ``"sushiswap"``, ``"shapeshift"``, ``"uphold"``, ``"bitpanda"``, ``"bisq"``, ``"ftxus"`` and ``"okx"``.

.. note::
   In the columns where an asset is expected you will need to use the identifier that such asset has in rotki otherwise the row won't be read.

.. note::

>    If at any point, you're confused as regards the csv format, feel free to send us a message on `Discord <https://discord.gg/aGCxHG7>`_.

Reviewer: ```suggestion
   If at any point, you're confused as regards to the csv format, feel free to send us a message on `Discord <https://discord.rotki.com/>`_.
```

.. [1] Suggestions will appear for this field based on the available data.
.. [2] The date filter has to be in the DD/MM/YYYY HH:mm:ss format. You can completely skip the time part or just the seconds part, thus making DD/MM/YYYY or DD/MM/YYYY HH:mm acceptable.

> History events

Reviewer: why move those?

Author: So I move this under "Historical Events" header, because I feel it's easier to find.
Previously it was under "Tracking accounts and balances"

<img width="308" alt="image" src="https://github.com/rotki/rotki/assets/26648140/83196db5-4ade-4eee-a90c-fc211b82f1a0">

- ``Event Type`` to ``Migration`` if it is a migration of assets from one protocol to another and you don't lose / gain anything from this event. For example when migrating from SAI to DAI.
- ``Event Type`` to ``Staking`` and ``Event Subtype`` to ``Deposit Asset`` if it is a staking event. For example staking in eth2 or in liquity.
- ``Event Type`` to ``Renew`` if it is a renewal of any subscription or service that you are paying for.
- ``Event Type`` to ``Informational`` if the event contains some useful information but it shouldn't be considered in accounting at all.

Events that have been modified will appear marked in the UI.

.. image:: images/customized_events.png
   :alt: Customized events in the UI
   :align: center

History events can be filtered if you have a premium subscription activated. You can filter by:

- Account (a tracked blockchain address)
- Time range
- Asset involved in the transaction
- Protocol that interacted in the transaction
- Location of the event (ethereum, optimism, kraken, etc.)
- Event type (deposit, withdrawal, etc.)
- Tx hash of a particualar transaction that you want to check

> - Index of an eth2 validator that you want to see events for

Reviewer: Keep it open for later

```suggestion
- Index of an eth2 validator that you want to see events for
- ... more
```

This can happen for many reasons. The asset may have been acquired in a non-supported exchange/protocol, some event not detected etc.

> The way to fix it is to add either a :ref:`manual trade<adding-manual-trade>` to tell rotki how you acquired that asset.

Reviewer: Also add  a link to how you can add history events

```suggestion
The way to fix it is to add either a :ref:`manual trade<adding-manual-trade>` to tell rotki how you acquired that asset or an acquisition history event.
```
